xbmc/filesystem/File.cpp

```diff
@@ -254,6 +254,19 @@ bool CFile::Open(const std::string& strFileName, const unsigned int flags)
 
 bool CFile::Open(const CURL& file, const unsigned int flags)
 {
+  if (m_pFile)
+  {
+    if ((flags & READ_REOPEN) == 0)
+    {
+      CLog::Log(LOGERROR, "File::Open - already open: %s", file.GetRedacted().c_str());
+      throw;
```

Comment:
for that purpose you could place an assert(0). as long as catch can not be enforced, we should not use throw at all. such kind of things can spread like cancer.

recently I tracked down an issue caused by one of our silly throw statements. this was not trivial to spot when all you get is a log file.